components/ListServices/service.jsx

```diff
@@ -36,31 +37,35 @@ const Service = ({ account }) => {
   const [ethTokenAddress, setEthTokenAddress] = useState(null);
   const [isUpdating, setIsUpdating] = useState(false);
 
-  const id = get(router, 'query.id') || null;
+  const id = router?.query?.id;
 
   useEffect(() => {
     (async () => {
       if (account) {
-        setAllLoading(true);
-        setServiceInfo({});
-
-        const result = await getServiceDetails(id);
-        setAllLoading(false);
-        setServiceInfo(result);
-
-        // get token address for L1 only network
-        // because L2 network do not have token address
-        if (isL1OnlyNetwork(chainId)) {
-          const token = await getTokenAddressRequest(id);
-          setEthTokenAddress(token);
+        try {
+          setAllLoading(true);
+          setServiceInfo({});
+
+          const result = await getServiceDetails(id);
+          setAllLoading(false);
+          setServiceInfo(result);
+
+          // get token address for L1 only network
+          // because L2 network do not have token address
```

Comment:
Not to change in this PR – I know this is an old comment – but the language/naming is confusing. We don't support any L2s at all currently. What you mean is tokenomics-supported networks.